### ISSUE TITLE
Can't install sharetribe at step 7. of installation instructions

### ISSUE THREAD

**Issue by jakoberr** [sharetribe/sharetribe#1562]
I can't successfully setup sharetribe (just trying to do this for development purposes on OS X El Capitan) at step 7 in the installation instructions (namely: bundle rake exec db:schema:load).

I get the following error message(s):

This is basically what is given out from the terminal:

>Digest::Digest is deprecated; use Digest
>sh: line 1:  1093 Trace/BPT trap: 5       indexer 2>&1
>INFO Connecting to database specified by database.yml
>
>AssetSync enabled: false
>
>Using the following AssetSync configuration:
>aws_access_key_id:     Not available
>aws_secret_access_key: Not available
>fog_directory:         Not available
>fog_provider:          Not available
>
>
>AssetSync enabled: false
>
>Using the following AssetSync configuration:
>aws_access_key_id:     Not available
>aws_secret_access_key: Not available
>fog_directory:         Not available
>fog_provider:          Not available
>
>-- create_table("auth_tokens", {:force=>true})
>DEBUG    (0.7ms)  CREATE TABLE `auth_tokens` (`id` int(11) DEFAULT NULL auto_increment >PRIMARY KEY, `token` varchar(255), `token_type` varchar(255) DEFAULT 'unsubscribe', `person_id` >varchar(255), `expires_at` datetime, `usages_left` int(11), `last_use_attempt` datetime, `created_at` >datetime NOT NULL, `updated_at` datetime NOT NULL) ENGINE=InnoDB
>DEBUG Mysql2::Error: All parts of a PRIMARY KEY must be NOT NULL; if you need NULL in a key, >use UNIQUE instead: CREATE TABLE `auth_tokens` (`id` int(11) DEFAULT NULL auto_increment >PRIMARY KEY, `token` varchar(255), `token_type` varchar(255) DEFAULT 'unsubscribe', `person_id` >varchar(255), `expires_at` datetime, `usages_left` int(11), `last_use_attempt` datetime, `created_at` >datetime NOT NULL, `updated_at` datetime NOT NULL) ENGINE=InnoDB
>rake aborted!
>ActiveRecord::StatementInvalid: Mysql2::Error: All parts of a PRIMARY KEY must be NOT NULL; if you >need NULL in a key, use UNIQUE instead: CREATE TABLE `auth_tokens` (`id` int(11) DEFAULT NULL >auto_increment PRIMARY KEY, `token` varchar(255), `token_type` varchar(255) DEFAULT >'unsubscribe', `person_id` varchar(255), `expires_at` datetime, `usages_left` int(11), `last_use_attempt` >datetime, `created_at` datetime NOT NULL, `updated_at` datetime NOT NULL) ENGINE=InnoDB
>/Users/NeuerBenutzer/.rvm/gems/ruby-2.1.2/gems/activerecord->3.2.21/lib/active_record/connection_adapters/abstract_mysql_adapter.rb:245:in `query'
>/Users/NeuerBenutzer/.rvm/gems/ruby-2.1.2/gems/activerecord->3.2.21/lib/active_record/connection_adapters/abstract_mysql_adapter.rb:245:in `block in execute'
>/Users/NeuerBenutzer/.rvm/gems/ruby-2.1.2/gems/activerecord->3.2.21/lib/active_record/connection_adapters/abstract_adapter.rb:280:in `block in log'
>/Users/NeuerBenutzer/.rvm/gems/ruby-2.1.2/gems/activesupport->3.2.21/lib/active_support/notifications/instrumenter.rb:20:in `instrument'
>/Users/NeuerBenutzer/.rvm/gems/ruby-2.1.2/gems/activerecord->3.2.21/lib/active_record/connection_adapters/abstract_adapter.rb:275:in `log'
>/Users/NeuerBenutzer/.rvm/gems/ruby-2.1.2/gems/activerecord->3.2.21/lib/active_record/connection_adapters/abstract_mysql_adapter.rb:245:in `execute'
>/Users/NeuerBenutzer/.rvm/gems/ruby-2.1.2/gems/activerecord->3.2.21/lib/active_record/connection_adapters/mysql2_adapter.rb:213:in `execute'
>/Users/NeuerBenutzer/.rvm/gems/ruby-2.1.2/gems/activerecord->3.2.21/lib/active_record/connection_adapters/abstract/schema_statements.rb:170:in `create_table'
>/Users/NeuerBenutzer/.rvm/gems/ruby-2.1.2/gems/activerecord->3.2.21/lib/active_record/connection_adapters/abstract_mysql_adapter.rb:434:in `create_table'
>/Users/NeuerBenutzer/.rvm/gems/ruby-2.1.2/gems/activerecord->3.2.21/lib/active_record/migration.rb:466:in `block in method_missing'
>/Users/NeuerBenutzer/.rvm/gems/ruby-2.1.2/gems/activerecord->3.2.21/lib/active_record/migration.rb:438:in `block in say_with_time'
>/Users/NeuerBenutzer/.rvm/gems/ruby-2.1.2/gems/activerecord->3.2.21/lib/active_record/migration.rb:438:in `say_with_time'
>/Users/NeuerBenutzer/.rvm/gems/ruby-2.1.2/gems/activerecord->3.2.21/lib/active_record/migration.rb:458:in `method_missing'
>/Users/NeuerBenutzer/Desktop/rubygems-2.5.0/sharetribe/db/schema.rb:16:in `block in <top >(required)>'
/Users/NeuerBenutzer/.rvm/gems/ruby-2.1.2/gems/activerecord-3.2.21/lib/active_record/schema.rb:50:in `instance_eval'
/Users/NeuerBenutzer/.rvm/gems/ruby-2.1.2/gems/activerecord-3.2.21/lib/active_record/schema.rb:50:in `define'
/Users/NeuerBenutzer/Desktop/rubygems-2.5.0/sharetribe/db/schema.rb:14:in `<top (required)>'
/Users/NeuerBenutzer/.rvm/gems/ruby-2.1.2/gems/zeus-0.15.1/lib/zeus/load_tracking.rb:50:in `load'
/Users/NeuerBenutzer/.rvm/gems/ruby-2.1.2/gems/zeus-0.15.1/lib/zeus/load_tracking.rb:50:in `load'
/Users/NeuerBenutzer/.rvm/gems/ruby-2.1.2/gems/zeus-0.15.1/lib/zeus/load_tracking.rb:43:in `load'
/Users/NeuerBenutzer/.rvm/gems/ruby-2.1.2/gems/activesupport-3.2.21/lib/active_support/dependencies.rb:245:in `block in load'
/Users/NeuerBenutzer/.rvm/gems/ruby-2.1.2/gems/activesupport-3.2.21/lib/active_support/dependencies.rb:236:in `load_dependency'
/Users/NeuerBenutzer/.rvm/gems/ruby-2.1.2/gems/activesupport-3.2.21/lib/active_support/dependencies.rb:245:in `load'
/Users/NeuerBenutzer/.rvm/gems/ruby-2.1.2/gems/activerecord-3.2.21/lib/active_record/railties/databases.rake:402:in `block (3 levels) in <top (required)>'
/Users/NeuerBenutzer/.rvm/gems/ruby-2.1.2/bin/ruby_executable_hooks:15:in `eval'
/Users/NeuerBenutzer/.rvm/gems/ruby-2.1.2/bin/ruby_executable_hooks:15:in `<main>'
Mysql2::Error: All parts of a PRIMARY KEY must be NOT NULL; if you need NULL in a key, use UNIQUE instead
/Users/NeuerBenutzer/.rvm/gems/ruby-2.1.2/gems/activerecord-3.2.21/lib/active_record/connection_adapters/abstract_mysql_adapter.rb:245:in `query'
/Users/NeuerBenutzer/.rvm/gems/ruby-2.1.2/gems/activerecord-3.2.21/lib/active_record/connection_adapters/abstract_mysql_adapter.rb:245:in `block in execute'
/Users/NeuerBenutzer/.rvm/gems/ruby-2.1.2/gems/activerecord-3.2.21/lib/active_record/connection_adapters/abstract_adapter.rb:280:in `block in log'
/Users/NeuerBenutzer/.rvm/gems/ruby-2.1.2/gems/activesupport-3.2.21/lib/active_support/notifications/instrumenter.rb:20:in `instrument'
/Users/NeuerBenutzer/.rvm/gems/ruby-2.1.2/gems/activerecord-3.2.21/lib/active_record/connection_adapters/abstract_adapter.rb:275:in `log'
/Users/NeuerBenutzer/.rvm/gems/ruby-2.1.2/gems/activerecord-3.2.21/lib/active_record/connection_adapters/abstract_mysql_adapter.rb:245:in `execute'
/Users/NeuerBenutzer/.rvm/gems/ruby-2.1.2/gems/activerecord-3.2.21/lib/active_record/connection_adapters/mysql2_adapter.rb:213:in `execute'
/Users/NeuerBenutzer/.rvm/gems/ruby-2.1.2/gems/activerecord-3.2.21/lib/active_record/connection_adapters/abstract/schema_statements.rb:170:in `create_table'
/Users/NeuerBenutzer/.rvm/gems/ruby-2.1.2/gems/activerecord-3.2.21/lib/active_record/connection_adapters/abstract_mysql_adapter.rb:434:in `create_table'
/Users/NeuerBenutzer/.rvm/gems/ruby-2.1.2/gems/activerecord-3.2.21/lib/active_record/migration.rb:466:in `block in method_missing'
/Users/NeuerBenutzer/.rvm/gems/ruby-2.1.2/gems/activerecord-3.2.21/lib/active_record/migration.rb:438:in `block in say_with_time'
/Users/NeuerBenutzer/.rvm/gems/ruby-2.1.2/gems/activerecord-3.2.21/lib/active_record/migration.rb:438:in `say_with_time'
/Users/NeuerBenutzer/.rvm/gems/ruby-2.1.2/gems/activerecord-3.2.21/lib/active_record/migration.rb:458:in `method_missing'
/Users/NeuerBenutzer/Desktop/rubygems-2.5.0/sharetribe/db/schema.rb:16:in `block in <top (required)>'
/Users/NeuerBenutzer/.rvm/gems/ruby-2.1.2/gems/activerecord-3.2.21/lib/active_record/schema.rb:50:in `instance_eval'
/Users/NeuerBenutzer/.rvm/gems/ruby-2.1.2/gems/activerecord-3.2.21/lib/active_record/schema.rb:50:in `define'
/Users/NeuerBenutzer/Desktop/rubygems-2.5.0/sharetribe/db/schema.rb:14:in `<top (required)>'
/Users/NeuerBenutzer/.rvm/gems/ruby-2.1.2/gems/zeus-0.15.1/lib/zeus/load_tracking.rb:50:in `load'
/Users/NeuerBenutzer/.rvm/gems/ruby-2.1.2/gems/zeus-0.15.1/lib/zeus/load_tracking.rb:50:in `load'
/Users/NeuerBenutzer/.rvm/gems/ruby-2.1.2/gems/zeus-0.15.1/lib/zeus/load_tracking.rb:43:in `load'
/Users/NeuerBenutzer/.rvm/gems/ruby-2.1.2/gems/activesupport-3.2.21/lib/active_support/dependencies.rb:245:in `block in load'
/Users/NeuerBenutzer/.rvm/gems/ruby-2.1.2/gems/activesupport-3.2.21/lib/active_support/dependencies.rb:236:in `load_dependency'
/Users/NeuerBenutzer/.rvm/gems/ruby-2.1.2/gems/activesupport-3.2.21/lib/active_support/dependencies.rb:245:in `load'
/Users/NeuerBenutzer/.rvm/gems/ruby-2.1.2/gems/activerecord-3.2.21/lib/active_record/railties/databases.rake:402:in `block (3 levels) in <top (required)>'
/Users/NeuerBenutzer/.rvm/gems/ruby-2.1.2/bin/ruby_executable_hooks:15:in `eval'
/Users/NeuerBenutzer/.rvm/gems/ruby-2.1.2/bin/ruby_executable_hooks:15:in `<main>'
Tasks: TOP => db:schema:load
(See full trace by running task with --trace)

**Comment by rap1ds** [comment]
@jakoberr Thanks for reporting the issue!

What's your MySQL version? I'm guessing that it's 5.7+, can you confirm that?

From MySQL 5.7 Changelog:

http://dev.mysql.com/doc/relnotes/mysql/5.7/en/news-5-7-3.html

```
Columns in a PRIMARY KEY must be NOT NULL, but if declared explicitly as NULL produced no error. Now an error occurs. For example, a statement such as CREATE TABLE t (i INT NULL PRIMARY KEY) is rejected. The same occurs for similar ALTER TABLE statements. (Bug #13995622, Bug #66987, Bug #15967545, Bug #16545198)
```

**Comment by rap1ds** [comment]
See also https://github.com/rails/rails/pull/13247

**Comment by jakoberr** [comment]
@rap1ds Thank you for your quick answer :) 

MySQL version is 5.6.27, for osx10.11 (x86_64), I've already tried to fix around with the patch mentioned in the thread, you've linked to. 
This changes the error from the above mentioned statement to:

>NameError: uninitialized constant >ActiveRecord::ConnectionAdapters::MysqlAdapter::NATIVE_DATABASE_TYPES

(Also tried the variations, mentioned in the thread, you've linked to, e.g. AbstractMysqlAdapter instead of MysqlAdapter, but none of them worked)

**Comment by jakoberr** [comment]
Solved this issue with installing MySQL 5.7 and fixing the abstract_mysql_adapter.rb in my active-record-3.2.21 folder.

Changed this:

>NATIVE_DATABASE_TYPES = {
>       :primary_key => "int(11) Default NULL auto_increment PRIMARY KEY",
>       [...]
>      }

To this:

>NATIVE_DATABASE_TYPES = {
>       :primary_key => "int(11) auto_increment PRIMARY KEY",
>       [...]
>      }

Not completely finished with the installation, but this particular issue is solved now.

**Comment by williamrussellajb** [comment]
This fixed it for me as well. 
:primary_key => "int(11) auto_increment PRIMARY KEY",
